dspace-server-webapp/src/main/java/org/dspace/app/rest/BitstreamRestController.java

```diff
@@ -131,32 +133,37 @@ public ResponseEntity retrieve(@PathVariable UUID uuid, HttpServletResponse resp
                     bit));
         }
 
-        // Pipe the bits
-        InputStream is = bitstreamTuple.getLeft();
         try {
-            HttpHeadersInitializer httpHeadersInitializer = HttpHeadersInitializer
-                    .fromInputStream(is)
-                    .withBufferSize(BUFFER_SIZE)
-                    .withFileName(name)
-                    .withLength(bitstreamTuple.getRight())
-                    .withChecksum(bit.getChecksum())
-                    .withMimetype(mimetype)
-                    .with(request)
-                    .with(response);
+            HttpHeadersInitializer httpHeadersInitializer = new HttpHeadersInitializer()
+                .withBufferSize(BUFFER_SIZE)
+                .withFileName(name)
+                .withLength(bit.getSizeBytes())
```

Comment:
the response size will be different from the bitstream size if the cover page is used

Reply:
@KevinVdV : I think @abollini is correct in both his comments...these code changes need to be using your new `getCitedDocumentLength()` method whenever the cover page is enabled.  It looks like that may only require a minor refactor to your PR though... but, I'll wait to review/test this until that minor cleanup is complete.